src/VkBootstrap.cpp

```diff
@@ -1664,6 +1664,11 @@ VkPresentModeKHR find_present_mode(std::vector<VkPresentModeKHR> const& availabl
 	return VK_PRESENT_MODE_FIFO_KHR;
 }
 
+bool is_unextended_present_mode(VkPresentModeKHR present_mode) {
+	return (present_mode == VK_PRESENT_MODE_IMMEDIATE_KHR) || (present_mode == VK_PRESENT_MODE_MAILBOX_KHR) ||
+	       (present_mode == VK_PRESENT_MODE_FIFO_KHR) || (present_mode == VK_PRESENT_MODE_FIFO_RELAXED_KHR);
+}
```

Comment:
Why do we need this check here? Does the present mode not being one of the 4 Vulkan 1.0 present modes pose additional usage/format feature requirements? If so, that should be documented in the implementation code that uses this function.

Reply:
It is because of the Note explanation on [VkSurfaceCapabilitiesKHR](https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/VkSurfaceCapabilitiesKHR.html) stating that the `supportedUsageFlags` only applies to those present modes, and not the shared ones. One would have to check if `VkSharedPresentSurfaceCapabilitiesKHR ` extension is enabled to perform further validation, but it goes out of scope of the work I wanted to do for this PR. I felt it would be a nice "simple" addition, but I didn't want to do the heavier work the validation layers do.

On next commit, I will add comments on this function so that it is clear. I could remove this validation because the layers do it anyway, but I felt that it would improve the library at small cost.

Reply:
That makes sense. Thanks for the clarification (and yeah, its kinda annoying how there are present modes for desktop and present modes for mobile, with different requirements on each...